### ISSUE TITLE
CommonVoice data splitting tr/dev/test

### ISSUE THREAD

**Issue by TParcollet** [espnet/espnet#2104]
Hi there !

Just a quick question on the data preparation of the CommonVoice recipe. From my understanding the Dev and Test are obtained from a split on the validated.tsv file (full set of audio files). Nonetheless, dev.tsv and test.tsv are also given by CommonVoice, why not using these files that already do not contain any duplicate in terms of speakers ?

Thanks !

**Comment by sw005320** [comment]
As you may know, the original commonvoice training/dev/eval split has a (very critical) problem for e2e research.
The same prompt is shared across training/dev/eval and e2e models can easily get 100% accuracy due to this overlap since e2e models also learn the LM context (which is OK for hybrid acoustic model training BTW).
(I was reporting it to Mozzila, already and sorry if they already fix the issue. I don't track it)

The same issue happens in the voxforge data.
We should be very careful when people report the commonvoice/voxforge e2e ASR performance, but this is not their fault but just a corpus design issue.

So, our decision is to make our own split.
We could still split them by considering both speakers and transcripts, but it makes the script very complicated, and also limits the variation of the training data transcript very small (again this is not appropriate for e2e research)
I think we already put such comments on the script, but probably it's better to emphasize it more.

**Comment by TParcollet** [comment]
I see, but prompts from the test / dev / train sets aren't uttered by the same speakers right?

**Comment by sw005320** [comment]
Yes for the original division.
Again, we actually tried to remove the sentences in the training data that appeared in the dev/evaluation data by keeping the original divisions as much as possible.
But it turns out that it loses a lot of training utterances in some languages, and decided the current solution (not perfect but open/close speaker conditions are not very important in these days).

**Comment by TParcollet** [comment]
Do you mean that validated.tsv contains couples of sentences+speakers that are also contained in the official dev and test?

**Comment by sw005320** [comment]
no only sentences

**Comment by TParcollet** [comment]
Alright thanks for the clarifications. I am unsure if this truly is a problem though, since it is most likely to happen in realistic conditions.

**Comment by sw005320** [comment]
This is OK for a real conversation since it happens naturally in the small portion.
But it is a problem in the prompt based read speech. commonvoice has relatively very large portions of unnatural text overlap with unrealistic occurrences (just sampling I guess). 

Again, I could get really really good WER in the commonvoice task with the original devision, but this is simply over-training (we just got 2% WER when we did not care about it. Almost perfect, and then it turns out that the above issue). I could get even better numbers by training it more and more, but the result is not meaningful...
So, I don't recommend you to use the original split anyway for the e2e setup.

**Comment by TParcollet** [comment]
Well for now I use a custom "clean" subset in the same manner as for librispeech, and it doesn't contain any sentence in common with the test and dev. But thanks for the pointer. I'll consider that. But then, we need to fix or release a new fixed test set so we can compare our different models ...

**Comment by sw005320** [comment]
Of course, it is no problem to use the original split or whatever for the training only purpose and apply it to the other test sets.
That is a normal use case and their main intention is also like that. So, there is no issue for their main purpose. It would be an issue for researchers to report the numbers by only using commonvoice data setup.

**Comment by sw005320** [comment]
> Well for now I use a custom "clean" subset in the same manner as for librispeech, and it doesn't contain any sentence in common with the test and dev. But thanks for the pointer. I'll consider that. But then, we need to fix or release a new fixed test set so we can compare our different models ...

I agree.

**Comment by sw005320** [comment]
Also, probably our information is too old(?). It was one year ago. Again, I asked them to fix it but I did not get any response, but probably they might fix it. If so, we should use the original split. @ftshijt, could you review their newest release?

**Comment by TParcollet** [comment]
The only new information I got: there is a train.tsv file that corresponds to the validated.tsv but without duplicate prompts. I don't know about the test and dev. Please keep me updated :D

**Comment by TParcollet** [comment]
Just in case, this is the official "data preparer" : https://github.com/mozilla/CorporaCreator

**Comment by sw005320** [comment]
Thanks! We only access to the original release directory. Probably we should follow this one. @ftshijt, could you check it?

**Comment by TParcollet** [comment]
The official data release follows this script, this is just a pointer given by Josh to me.

**Comment by sw005320** [comment]
I see.

**Comment by ftshijt** [comment]
Thanks for bringing out the issue and appreciate your information @TParcollet . I just took a look at the commonvoice release. It seems that there was a big update in Dec 2019 (most corpora have expended). The current ESPNet recipe is based on the release in Aug 2019, a bit out-of-date. I will update the recipe & pre-trained models and also see if the train-test split is usable for our e2e system.

**Comment by TParcollet** [comment]
@ftshijt Josh Meyer mentioned that all the train.tsv only contain unique sentences with respect to the dev / test and itself. I think that the condition on itself is too restrictive and might remove a lot of audio files. I'll try to verify and confirm you that.
